airflow/providers/mongo/hooks/mongo.py

```diff
@@ -46,15 +46,15 @@ class MongoHook(BaseHook):
         when connecting to MongoDB.
     """
 
-    conn_name_attr = "conn_id"
+    conn_name_attr = "mongo_conn_id"
     default_conn_name = "mongo_default"
     conn_type = "mongo"
     hook_name = "MongoDB"
 
-    def __init__(self, conn_id: str = default_conn_name, *args, **kwargs) -> None:
+    def __init__(self, mongo_conn_id: str = default_conn_name, *args, **kwargs) -> None:
```

Comment:
This add breaking changes in case if user provide arguments as keyword:

```python

MongoHook(conn_id="awesome-conn-id")
```

Reply:
@Wesseldr Thanks for the contribution and your first PR! Definitely the docstring for the hook doesn't match what's _actually_ required for use.

+1 on what @Taragolis mentioned.

We try to be very cognizant when thinking about backwards compat for users and try not to introduce breaking changes if avoidable. We really don't want users to upgrade a provider version and suddenly DAGs begin to break. In this situation, if the parameter name is changing, a `DeprecationWarning` should be raised that `conn_id` is deprecated and `mongo_conn_id` should be used. You can find a number of examples in other providers' hooks where a parameter is deprecated for reference on how to do this.

For the deprecation period, the hook should accept both `conn_id` and `mongo_conn_id` to maintain backwards compat and ultimately resolve to the `self.mongo_conn_id` value. Also the [`MongoSensor` should _ideally_ be updated to call this hook with keyword args](https://github.com/apache/airflow/blob/12c3c39d1a816c99c626fe4c650e88cf7b1cc1bc/airflow/providers/mongo/sensors/mongo.py#L62).

The unit tests for the provider need to be updated accordingly too.

We're more than happy to help out if you are stuck. You can find us on Airflow Slack or we can communicate in this PR directly too.